revalidation/initial_contacts/_02_icd_result_generation_no_exc.py

```diff
@@ -0,0 +1,320 @@
+        ] = None,
+        dp_group: Optional[tuple[str, ...]] = None,
+        current_gs: Region = None,
+        **_: Unpack[dict[str, Any]],
+    ) -> Self:
+        """ "Run" the algorithm."""
+        assert (
+            measurement_condition is not None
+        ), "measurement_condition must be provided"
+        assert dp_group is not None, "dp_group must be provided"
+
+        cached_load_old_icd_results = hybrid_cache(lru_cache_maxsize=1)(
+            load_old_icd_results
+        )
+
+        all_results = cached_load_old_icd_results(
+            self.base_result_folder
+            / measurement_condition
+            / f"{self.old_algo_name}.csv"
+        )
+
+        unique_label = dp_group[:-2]
+        gs_start = current_gs.start
+        try:
+            ic_results = (
+                all_results.loc[unique_label]
+                .query(
+                    "start == @gs_start | start == @gs_start + 1 | start == @gs_start - 1"
+                )
+                .copy()
+            )
+            ic = ic_results["ic_list_rel_to_wb"].apply(pd.eval).iloc[0]
+            ic_list = pd.DataFrame(
+                {"ic": ic},
+            )
+            ic_list.index.names = ["step_id"]
+        except:
+            # returns an empty dataframe for all exceptions (missing algo result for the data or missing data)
+            ic_list = pd.DataFrame(columns=["ic"])
+
+        self.ic_list_ = ic_list
+        return self
+
+
+# %%
+# Setting up the algorithms
+# -------------------------
+# We use the :class:`~mobgap.initial_contacts.pipeline.IcdEmulationPipeline` to run the algorithms.
+# We create an instance of this pipeline for each algorithm we want to evaluate and store them in a dictionary.
+# The key is used to identify the algorithm in the results and used as folder name to store the results.
+#
+# .. note:: Set up your environment variables to point to the correct paths.
+#    The easiest way to do this is to create a `.env` file in the root of the repository with the following content.
+#    You need the paths to the root folder of the TVS dataset `MOBGAP_TVS_DATASET_PATH` and the path where revalidation
+#    results should be stored `MOBGAP_VALIDATION_DATA_PATH`.
+#    The path to the cache directory `MOBGAP_CACHE_DIR_PATH` is optional, when you don't want to store the memory cache
+#    in the default location.
+from mobgap.initial_contacts.pipeline import IcdEmulationPipeline
+from mobgap.utils.misc import get_env_var
+
+matlab_algo_result_path = (
+    Path(get_env_var("MOBGAP_VALIDATION_DATA_PATH")) / "_extracted_results/icd"
+)
+
+pipelines = {}
+for matlab_algo_name in [
+    "Ani_McCamley",
+]:
+    pipelines[f"matlab_{matlab_algo_name}"] = IcdEmulationPipeline(
+        DummyIcdAlgo(
+            matlab_algo_name, base_result_folder=matlab_algo_result_path
+        )
+    )
+
+# %%
+# For the reimplemented algorithm, we set up version with different default presets.
```

Comment:
For the ICD algos, we don't really have different presets. Can you update the comment?